Python/ast.c

```diff
@@ -126,23 +128,25 @@ validate_constant(PyObject *value)
         PyObject *it;
 
         it = PyObject_GetIter(value);
-        if (it == NULL)
-            return 0;
+        if (it == NULL) {
+            goto invalid;
```

Comment:
Why not keep the old code?

Reply:
What do you mean by keeping the old code? In the old code, the exception was setted inside of validate_expr if this function returns 0, now it is setted inside of this function so we need to go and set the exception and return 0.

Reply:
Oh, I see now. Makes sense, thanks.